tests/fixtures/optimizer/qualify_columns.sql

```diff
@@ -0,0 +1,127 @@
+--------------------------------------
+-- Qualify columns
+--------------------------------------
+SELECT a FROM x;
+SELECT "x"."a" AS "a" FROM x;
+
+SELECT a AS a FROM x;
+SELECT "x"."a" AS "a" FROM x;
+
+SELECT x.a FROM x;
+SELECT "x"."a" AS "a" FROM x;
+
+SELECT "x"."a" AS "a" FROM x;
+SELECT "x"."a" AS "a" FROM x;
+
+SELECT a AS b FROM x;
+SELECT "x"."a" AS "b" FROM x;
+
+SELECT 1, 2 FROM x;
+SELECT 1 AS "_col_0", 2 AS "_col_1" FROM x;
+
+--------------------------------------
+-- Derived tables
+--------------------------------------
+SELECT y.a AS a FROM (SELECT x.a AS a FROM x) AS y;
+SELECT "y"."a" AS "a" FROM (SELECT "x"."a" AS "a" FROM x) AS y;
+
+SELECT y.a AS a FROM (SELECT x.a AS a FROM x) AS y(a);
+SELECT "y"."a" AS "a" FROM (SELECT "x"."a" AS "a" FROM x) AS y;
+
+SELECT y.a AS a FROM (SELECT x.a AS a, x.b AS b FROM x) AS y(c);
+SELECT "y"."a" AS "a" FROM (SELECT "x"."a" AS "c", "x"."b" AS "b" FROM x) AS y;
+
+SELECT a FROM (SELECT a FROM x) y;
+SELECT "y"."a" AS "a" FROM (SELECT "x"."a" AS "a" FROM x) AS y;
+
+SELECT a FROM (SELECT a AS a FROM x);
+SELECT "_q_0"."a" AS "a" FROM (SELECT "x"."a" AS "a" FROM x) AS "_q_0";
+
+SELECT a FROM (SELECT a FROM (SELECT a FROM x));
+SELECT "_q_0"."a" AS "a" FROM (SELECT "_q_1"."a" AS "a" FROM (SELECT "x"."a" AS "a" FROM x) AS "_q_1") AS "_q_0";
```

Comment:
i notice your context is a global context and not scoped, since there wasn't a reason to increment _q_1 here, but that's fine, just an observation

Reply:
Ah, right, the counter is global. That's a good point. I cant reset the counter.